src/mgmt/tests/mocks.c

```diff
@@ -272,41 +272,41 @@ ds_mgmt_free_pool_list(struct mgmt_list_pools_one **poolsp, uint64_t len)
 	}
 }
 
-int				ds_mgmt_set_prop_return;
+int		ds_mgmt_pool_set_prop_return;
+daos_prop_t	*ds_mgmt_pool_set_prop_prop;
+daos_prop_t	*ds_mgmt_pool_set_prop_result;
+void		*ds_mgmt_pool_set_prop_result_ptr;
 int
 ds_mgmt_pool_set_prop(uuid_t pool_uuid, daos_prop_t *prop,
 		      daos_prop_t **result)
 {
-	daos_prop_t *res_prop;
-	size_t i;
-
-	if (ds_mgmt_set_prop_return != 0)
-		return ds_mgmt_set_prop_return;
+	if (prop != NULL)
+		ds_mgmt_pool_set_prop_prop = daos_prop_dup(prop, true);
+	ds_mgmt_pool_set_prop_result_ptr = (void *)result;
 
-	res_prop = daos_prop_alloc(prop->dpp_nr);
-	if (res_prop == NULL)
-		return -DER_NOMEM;
+	if (result != NULL && ds_mgmt_pool_set_prop_result != NULL) {
+		size_t len = ds_mgmt_pool_set_prop_result->dpp_nr;
 
-	for (i = 0; i < prop->dpp_nr; i++) {
-		res_prop->dpp_entries[i].dpe_type =
-			prop->dpp_entries[i].dpe_type;
-		res_prop->dpp_entries[i].dpe_val =
-			prop->dpp_entries[i].dpe_val;
+		*result = daos_prop_alloc(len);
+		daos_prop_copy(*result, ds_mgmt_pool_set_prop_result);
 	}
 
-	*result = res_prop;
-	return 0;
+	return ds_mgmt_pool_set_prop_return;
 }
 
 void
 mock_ds_mgmt_pool_set_prop_setup(void)
 {
-	ds_mgmt_set_prop_return = 0;
+	ds_mgmt_pool_set_prop_return = 0;
+	ds_mgmt_pool_set_prop_prop = NULL;
+	ds_mgmt_pool_set_prop_result = NULL;
+	ds_mgmt_pool_set_prop_result_ptr = NULL;
 }
 
 void
 mock_ds_mgmt_pool_set_prop_teardown(void)
 {
+	daos_prop_free(ds_mgmt_pool_set_prop_result);
```

Comment:
Also need to cleanup ds_mgmt_pool_set_prop_prop since daos_prop_dup allocates it.

Reply:
https://jira.hpdd.intel.com/browse/DAOS-3938